### PR TITLE
Add CI security action

### PR DESCRIPTION
**Pull request by ocaisa**
There is _a lot_ going on in CI in this repo, permissions need to be pinned down, and we also need to check if all these external actions are really needed

### CODE REVIEW

File: .github/workflows/build-test-release-client-packages.yml [hunk 1]
```diff
@@ -10,13 +10,16 @@ on:
     branches:
       - main
 
+# Declare default permissions as read only.
+permissions: read-all
```
Comment (ocaisa) [inline .github/workflows/build-test-release-client-packages.yml:14]: This one is likely to need additional permissions but I'm not sure what

File: .github/workflows/build-test-release-client-packages.yml [hunk 2]
```diff
@@ -287,7 +290,7 @@ jobs:
           ls -1 -R .
           echo ::set-output name=package_filenames::$(find ./build_artifacts -name cvmfs-config-eessi*)
 
-      - uses: meeDamian/github-release@2.0
+      - uses: meeDamian/github-release@7ae19492500104f636b3fee4d8103af0fed36c8e # v2.0.3
```
Comment (ocaisa) [inline .github/workflows/build-test-release-client-packages.yml:293]: This has surely got to require some permissions

File: .github/workflows/build-test-release-client-packages.yml [hunk 3]
```diff
@@ -324,14 +327,14 @@ jobs:
           echo ::set-output name=package_filenames::$(find ./build_artifacts -name cvmfs-config-eessi*latest*)
 
       - name: Delete any previous tag/release named "latest"
-        uses: dev-drprasad/delete-tag-and-release@v0.1.3
+        uses: dev-drprasad/delete-tag-and-release@085c6969f18bad0de1b9f3fe6692a3cd01f64fe5 # v0.2.0
```
Comment (ocaisa) [inline .github/workflows/build-test-release-client-packages.yml:330]: This too has got to require permissions

File: .github/workflows/build-test-release-client-packages.yml (hunk 2 above)
Comment (ocaisa) [reply .github/workflows/build-test-release-client-packages.yml:293]: The `release` and `release-latest` actions are only run in certain scenarios, so can't test this in a PR

**Comment by ocaisa** [comment]
@bedroge This should be ready for review now

File: .github/workflows/build-test-release-client-packages.yml (hunk 2 above)
Comment (bedroge) [reply .github/workflows/build-test-release-client-packages.yml:295]: I assume `contents: write` should be sufficient for this, but can't find it in the docs, so let's just try it out.

**Review by bedroge — APPROVED** [review]
lgtm

**Pull request merged**